simple_salesforce/util.py

```diff
@@ -34,4 +34,14 @@ def date_to_iso8601(date):
 
 class SalesforceError(Exception):
     """Base Salesforce API exception"""
-    pass
+
+    message = 'Unknown error occurred for {url}. Response content: {content}'
+
+    def __init__(self, url, status, resource_name, content):
+        self.url = url
+        self.status = status
+        self.resource_name = resource_name
+        self.content = content
+
+    def __str__(self):
```

Comment:
Yeah I've never been sure how Python 3 did exception message stringification. Closest thing I could find was https://docs.python.org/3.4/tutorial/errors.html#user-defined-exceptions but it's using `__str__` too!